### PR TITLE
Fix null event names and event properties 

### PR DESCRIPTION
**Pull request by daniloercoli**
This PR fix an issue where passing an empty user to the library generates a bad crash.
I've also added other checks over the properties passed to the event, so the library should not crash if an invalid input is given.

- Also modernized gradle, and SDK.

Version number is also bumped, so we can release a new version of the library when this is merged to develop.

cc @maxme

### CODE REVIEW

File: AutomatticTracks/src/main/java/com/automattic/android/tracks/datasets/EventTable.java [hunk 1]
```diff
@@ -97,11 +97,13 @@ public static void insertEvent(Context ctx, Event event) {
             }
 
             stmt.bindLong(8, event.getTimeStamp());
-            stmt.bindLong(9,  event.getRetryCount());
+            stmt.bindLong(9, event.getRetryCount());
 
             stmt.execute();
 
             db.setTransactionSuccessful();
+        } catch(IllegalArgumentException e) {
```
Comment (maxme) [inline AutomatticTracks/src/main/java/com/automattic/android/tracks/datasets/EventTable.java]: I think we should add our checkstyle config in this project (there is a missing space between `catch` and `(`)

**Comment by daniloercoli** [comment]
Fixed the missing space. Yeah, we may want to add the checkstyle config. Maybe in another PR?

**Comment by maxme** [comment]
> Maybe in another PR?

Yeah definitely not in that one, opened a ticket https://github.com/Automattic/Automattic-Tracks-Android/issues/31

**Comment by maxme** [comment]
PR looks good, I'll publish a new version to bintray. We can update the dependency in wpandroid 7.2.

**Pull request merged**